components/pagination/Pagination.js

```diff
@@ -0,0 +1,128 @@
+    currentPage: PropTypes.number.isRequired,
+    onChange: PropTypes.func.isRequired,
+    prevPageText: PropTypes.string,
+    className: PropTypes.string,
+  };
+
+  static defaultProps = {
+    currentPage: 1,
+    inverse: false,
+    numItemsPerPage: 10,
+    maxNumPagesVisible: 7,
+  };
+
+  render() {
+    const {
+      className,
+      currentPage,
+      inverse,
+      maxNumPagesVisible,
+      nextPageText,
+      numItems,
+      numItemsPerPage,
+      onChange,
+      prevPageText,
+      ...others
+    } = this.props;
+
+    const numPages = Math.ceil(numItems / numItemsPerPage);
+
+    if (numPages < 2) {
+      return null;
+    }
+
+    const classNames = cx(theme['pagination'], { [theme['is-inverse']]: inverse }, className);
+    const iterator = [currentPage];
+
+    let counter = 0;
+
+    while (iterator.length < maxNumPagesVisible && iterator.length < numPages) {
+      if (counter % 2 === 0 && iterator[0] !== 1) {
+        iterator.unshift(iterator[0] - 1);
+      } else if (counter % 2 === 1 && iterator[iterator.length - 1] !== numPages) {
+        iterator.push(iterator[iterator.length - 1] + 1);
+      }
+
+      counter++;
+    }
+
+    // make sure the first and the last item are the first and the last page
+    iterator[0] = 1;
+    iterator[iterator.length - 1] = numPages;
+
+    // should we replace the 2nd and 2nd last by ellipsis?
+    if (iterator.length > 1 && iterator[1] !== 2) {
+      iterator[1] = 'ellipsis-left';
+    }
+    if (iterator.length > 1 && iterator[iterator.length - 2] !== numPages - 1) {
+      iterator[iterator.length - 2] = 'ellipsis-right';
+    }
+
+    return (
+      <Box data-teamleader-ui="pagination" className={classNames} element="nav" {...others}>
+        <ul className={theme['list']}>
+          {currentPage > 1 && (
+            <li className={theme['list-item']}>
+              <LinkButton
+                icon={<IconChevronLeftMediumOutline />}
+                inverse={inverse}
+                label={prevPageText}
+                onClick={() => onChange(currentPage - 1)}
+                size="small"
+              />
+            </li>
+          )}
+          {iterator.map(page => {
+            return (
+              <li key={page} className={theme['list-item']}>
+                {String(page).startsWith('ellipsis') ? (
+                  <TextSmall className={theme['ellipsis']} element="span">
+                    ...
+                  </TextSmall>
+                ) : (
+                  <LinkButton
+                    className={page === currentPage && theme['current']}
+                    inverse={inverse}
+                    label={page}
+                    onClick={() => onChange(page)}
+                    size="small"
+                  />
+                )}
+              </li>
+            );
+          })}
+          {currentPage < numPages && (
+            <li className={theme['list-item']}>
+              <LinkButton
```

Comment:
I've been thinking about this component, and with the current implementation, I actually can't do nothing in the invoicing module. 

In my case the pagination buttons have to be React router links, and not just the fixed LinkButton. So we need to be able to pass some custom element or something like that. 

Current implementation uses onClick and onChange, but in my case I don't really want this, but it might be necessary in other implementations. Not sure how we could solve this.